broker/setup_broker/setup_broker.sh

```diff
@@ -36,34 +36,101 @@ if [ "$continue_with_setup" != "y" ]; then
     exit
 fi
 
+# function used to define GCP resources; appends testid if needed
+define_GCP_resources() {
+    local base_name="$1"
+    local testid_suffix=""
+
+    if [ "$testid" != "False" ]; then
+        if [ "$base_name" = "$survey" ]; then
+            testid_suffix="_${testid}"  # complies with BigQuery naming conventions
+        else
+            testid_suffix="-${testid}"
+        fi
+    fi
+
+    echo "${base_name}${testid_suffix}"
+}
+
 #--- GCP resources used directly in this script
-broker_bucket="${PROJECT_ID}-${survey}-broker_files"
-bq_dataset="${survey}"
-# use test resources, if requested
-# (there must be a better way to do this)
-if [ "$testid" != "False" ]; then
-    broker_bucket="${broker_bucket}-${testid}"
-    bq_dataset="${bq_dataset}_${testid}"
-fi
+broker_bucket=$(define_GCP_resources "${PROJECT_ID}-${survey}-broker_files")
+bq_dataset=$(define_GCP_resources "${survey}")
+# topics and subscriptions involved in writing DIASource data to BigQuery
+topic_diasource=$(define_GCP_resources "${survey}-diasource")
+subscription_diasource="${topic_diasource}" # BigQuery subscription
+topic_diasource_deadletter=$(define_GCP_resources "${survey}-diasource-deadletter")
+subscription_diasource_deadletter="${topic_diasource_deadletter}"
+# topics and subscriptions involved in writing alert data to BigQuery
+topic_alert_data=$(define_GCP_resources "${survey}-alert-data") # needs a better name
```

Comment:
I appreciate the point that `-data` doesn't seem like it's adding much.  Is this topic meant really to be used only by BigQuery?  If so, do you think it would be fine to call something specific like `ztf-alert-bigquery-import`?